File: src/test/java/io/springacademy/shortener/RedisLinkRepositoryTest.java
package io.springacademy.shortener;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import reactor.test.StepVerifier;

@RunWith(SpringRunner.class)
@SpringBootTest
public class RedisLinkRepositoryTest {

    @Autowired
    private RedisLinkRepository repository;

    @Test
    public void shouldSaveLink() {
        Link link = new Link("some-key", "http://spring.io");
        StepVerifier.create(repository.save(link))
                    .expectNext(link)
                    .verifyComplete();
    }

    @Test
    public void shouldFindLinkById() {
        Link link = new Link("some-key", "http://spring.io");
        StepVerifier.create(repository.save(link)
                                      .flatMap(result -> repository.findById(result.getKey())))
                    .expectNext(link)
                    .verifyComplete();
    }

}

File: src/test/java/io/springacademy/shortener/RedisLinkRepositoryTest.kt
package io.springacademy.shortener

import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner
import reactor.test.test

@RunWith(SpringRunner::class)
@SpringBootTest
class RedisLinkRepositoryTest {

    @Autowired
    private lateinit var repository: RedisLinkRepository

    @Test
    fun shouldSaveLink() {
        val link = Link("some-key", "http://spring.io")
        repository.save(link)
            .test()
            .expectNext(link)
            .verifyComplete()
    }

    @Test
    fun shouldFindLinkById() {
        val link = Link("some-key", "http://spring.io")
        repository.save(link)
            .flatMap { (key) -> repository.findById(key) }
            .test()
            .expectNext(link)
            .verifyComplete()
    }

}
